src/receivers/custom-routes.ts

```diff
@@ -0,0 +1,48 @@
+import { IncomingMessage, ServerResponse } from 'http';
+import { CustomRouteInitializationError } from '../errors';
+
+export interface CustomRoute {
+  path: string;
+  method: string | string[];
+  handler: (req: IncomingMessage, res: ServerResponse) => void;
+}
+
+export interface ReceiverRoutes {
+  [key: string]: {
+    [key: string]: (req: IncomingMessage, res: ServerResponse) => void;
+  };
+}
+
+export function prepareRoutes(customRoutes: CustomRoute[]): ReceiverRoutes {
+  const routes: ReceiverRoutes = {};
+
+  validateCustomRoutes(customRoutes);
+
+  customRoutes.forEach((r) => {
+    const methodObj = Array.isArray(r.method) ?
+      r.method.reduce((o, key) => ({ ...o, [key.toUpperCase()]: r.handler }), {}) :
+      { [r.method.toUpperCase()]: r.handler };
+    routes[r.path] = routes[r.path] ? { ...routes[r.path], ...methodObj } : methodObj;
```

Comment:
Do we need to worry about duplicates here? I think they are handled here but just want to confirm.

Reply:
Great question. If a developer has included duplicates for the same `path` and `method`, the last one provided will be the one that is cached/used. I think that would be expected behavior, but happy to have that notion challenged